File: solutions/cpp/167/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        vector<int> ans;
        while (left < right) {
            int curr = numbers[left] + numbers[right];
            if (curr < target) {
                left++;
            } else if (curr > target) {
                right--;
            } else {
                ans.push_back(left + 1);
                ans.push_back(right + 1);
                return ans;
            }
        }
        return ans;
    }
}

File: solutions/golang/167/solution.go
func twoSum(numbers []int, target int) []int {
	left := 0
	right := len(numbers) - 1
	ans := make([]int, 2)
	for left < right {
		curr := numbers[left] + numbers[right]
		if curr == target {
			ans[0] = left + 1
			ans[1] = right + 1
			return ans
		} else if curr < target {
			left++
		} else {
			right--
		}
	}
	return ans
}